app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -46,13 +46,23 @@ def loa2_or_above?
   end
 
   def identity_not_verified?
-    UserDecorator.new(current_user).identity_not_verified?
+    !current_user.verified?
   end
 
   def active_identity
     current_user.last_identity
   end
 
+  def build_asserted_attributes(principal)
+    asserter = AttributeAsserter.new(principal, current_service_provider, saml_request)
+    asserter.build
```

Comment:
Does this return a value or have a side effect?  I guess its the later because the return value is ignored below